ocean_lib/web3_internal/contract_utils.py

```diff
@@ -51,24 +51,26 @@ def get_contracts_addresses_all_networks(config: dict):
 
 
 @enforce_types
-def get_contracts_addresses(config: dict) -> Optional[Dict[str, str]]:
+def get_contracts_addresses(config: dict, retries: int = 2) -> Optional[Dict[str, str]]:
     """Get addresses for given NETWORK_NAME, from info in ADDRESS_FILE"""
     network_name = config["NETWORK_NAME"]
     if network_name == "polygon-test":
         network_name = "mumbai"
 
     if network_name == "polygon-main":
         network_name = "polygon"
+
+    if retries == 0:
```

Comment:
there are several things I don't understand here:
- how do the retries work without sleeping? What makes the address suddenly appear between two subsequent instructions?
- I am concerned about the stop condition. What if a user calls this with retries=-3?
- The issue seems to be that even though there is an address file and barge marks itself as `ready`, it does not contain all keys. Please see the work I've done for showing the file itself. This would indicate there is a problem with barge itself, not ocean.py. Barge marks itself as `ready` before it is in fact ready. So what are we doing here: tying ocean.py functionality to a quirk in a dependency (barge). Even moreso, we are tying ocean.py functionality to a quirk in our dependency, when used inside our CI flows. Let's reconsider first the behaviour of barge and our CIs, not introduce such solutions

Reply:
I'll close this PR according to the last bullet, but keep the issue up.